### PR TITLE
1798 - update chainHead to use bestBlockAndState

### PR DESCRIPTION
**Pull request by rolfyone**
- changed /beacon/head to use bestBlockAndState also.

 - removed mocking from the test case to better show what is happening.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

### CODE REVIEW

File: data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java [hunk 1]
```diff
@@ -214,6 +207,10 @@ public boolean isFinalizedEpoch(final UnsignedLong epoch) {
   }
 
   public Optional<BeaconChainHead> getHeadState() {
-    return combinedChainDataClient.getHeadStateFromStore().map(BeaconChainHead::new);
+    if (!isStoreAvailable()) {
+      throw new ChainDataUnavailableException();
+    }
+    return recentChainData.getBestBlockAndState().map(BeaconChainHead::new);
+    //    return combinedChainDataClient.getHeadStateFromStore().map(BeaconChainHead::new);
```
Comment (ajsutton) [inline data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java]: nit: remove commented out code.

**Review by ajsutton — APPROVED** [review]
LGTM.

File: data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java (hunk 1 above)
Comment (rolfyone) [reply data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java]: gah! fair point!

**Review by mark-terry — APPROVED** [review]
LGTM.

**Pull request merged**